src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1434,16 +1438,24 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
         ///     This property forces all AST nodes that generated from this node
         ///     to be executed, even there is no change in AST nodes.
         /// </summary>
-        public virtual bool ForceReExecuteOfNode
+        [Flags]
+        public enum ExecutionHint
+        {
+            None = 0,
+            GenerateAst = 1,
+            ForceExecute = 3
+        }
+
+        private ExecutionHint executionHint = ExecutionHint.None;
+        public virtual ExecutionHint ExecutionHintFlag
         {
             get
             {
-                return forceReExec;
+                return executionHint;
             }
             set
             {
-                forceReExec = value;
-                RaisePropertyChanged("ForceReExecuteOfNode");
+                executionHint = value;
             }
         }
```

Comment:
This property may not need to exist, or it can exist as a read-only property. Our aim is to make it as easy as possible for external calls to deal with this value (see my comments for the overall pull request).